integration-tests/tests/src/tests/objects.ts

```diff
@@ -17,244 +17,1323 @@
+        expect(john._id.equals(_id)).equals(true);
+        expect(john.name).equals("John Doe");
+        expect(john.age).equals(43);
 
-      realm.write(() => {
-        realm.create<PersonWithId>(PersonSchemaWithId.name, {
+        const update: Partial<IPersonWithId> = {
           _id,
-          name: "John Doe",
-          age: 42,
+          name: "Mr. John Doe",
+        };
+
+        this.realm.write(() => {
+          this.realm.create<IPersonWithId>(PersonSchemaWithId.name, update, Realm.UpdateMode.Modified);
         });
+
+        expect(john._id.equals(_id)).equals(true);
+        expect(john.name).equals("Mr. John Doe");
+        expect(john.age).equals(43);
+
+        expect(() =>
+          this.realm.write(() => {
+            this.realm.create<IPersonWithId>(
+              PersonSchemaWithId.name,
+              { _id, name: "John Doe", age: 42 },
+              Realm.UpdateMode.Never,
+            );
+          }),
+        ).throws(
+          `Attempting to create an object of type '${PersonSchemaWithId.name}' with an existing primary key value '${_id}'.`,
+        );
+
+        // Expect only one instance of 'PersonSchemaWithId' in db after all updates
+        const persons = this.realm.objects(PersonSchemaWithId.name);
+        expect(persons.length).equals(1);
       });
+    });
+  });
 
-      const john = realm.objectForPrimaryKey<IPersonWithId>(PersonSchemaWithId.name, _id);
-      if (!john) throw new Error("Object not found");
+  describe("Class Model", () => {
+    describe("without primary key", () => {
+      openRealmBeforeEach({ schema: [Person] });
+      it("can be created", function (this: Mocha.Context & RealmContext) {
+        const john = this.realm.write(() => {
+          return this.realm.create(Person, {
+            name: "John Doe",
+            age: 42,
+          });
+        });
+        // Expect John to be the one and only result
+        const persons = this.realm.objects(Person);
+        expect(persons.length).equals(1);
+        const [firstPerson] = persons;
+        expect(firstPerson).deep.equals(john);
+        expect(firstPerson).instanceOf(Person);
+      });
+
+      it("can have it's properties read", () => {
+        const realm = new Realm({ schema: [Person] });
+        realm.write(() => {
+          const john = realm.create(Person, {
+            name: "John Doe",
+            age: 42,
+          });
 
-      expect(john).instanceOf(Realm.Object);
-      expect(john._id.equals(_id)).equals(true);
-      expect(john.name).equals("John Doe");
-      expect(john.age).equals(42);
+          expect(john.name).equals("John Doe");
+          expect(john.age).equals(42);
+        });
+      });
     });
 
-    it("can be updated", () => {
-      const realm = new Realm({ schema: [PersonSchemaWithId] });
-      const _id = new Realm.BSON.ObjectId();
+    describe("with primary key", () => {
+      openRealmBeforeEach({ schema: [PersonWithId] });
```

Comment:
I'm wondering if we should just inline define the schema in the `openRealmBeforeEach`.  This would make it clear at the top of the test what properties have been set without scrolling to the top of the test.  Possible refactoring/discussion

Reply:
yeah that is definitely a pro, my counterargument though is that if we inline the definition, then for methods like this
![image](https://user-images.githubusercontent.com/44238293/216290219-27a74566-4730-4290-96d2-5c355b868fef.png)
we would have to replace `PersonSchema.name` with its raw value (unless there's no other way to do that). 

Think I'm biased as Well since I use "gd" to go to definition via vim and "ctrl + o" to jump back to previous location in file